google-cloud-clients/google-cloud-bigquery/src/main/java/com/google/cloud/bigquery/DatasetId.java

```diff
@@ -47,12 +48,15 @@ private DatasetId(String project, String dataset) {
 
   /** Creates a dataset identity given project's and dataset's user-defined ids. */
   public static DatasetId of(String project, String dataset) {
-    return new DatasetId(checkNotNull(project), checkNotNull(dataset));
+    checkArgument(!isNullOrEmpty(project), "Provided project is null or empty");
```

Comment:
I think the message should be opposite something like "project id can't be null or empty"... just for consistency.

Reply:
You can leave it, I think in BigQuery you follow this type of comment